### ISSUE TITLE
Bogus Warning in lambda expression

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#367]
```
What steps will reproduce the problem?
1. Run the Null Checker on the class below
package test;

class Test {
    static void test(Iterable<? extends Thread> threads) {
        threads.forEach(thread -> System.out.println(thread));
    }
}

What is the expected output? What do you see instead?
Expected: no warning
Actual: test\Test.java:5: error: [lambda.param.type.incompatible] incompatible 
parameter types in lambda expression.
        threads.forEach(thread -> System.out.println(thread));
                        ^
  found   : ? extends @Initialized @NonNull Thread
  required: capture#214 of ? extends java.lang.Thread extends @Initialized @NonNull Thread
1 error

What version of the product are you using? On what operating system?
1.8.6/Windows 8/Java 8


```

Original issue reported on code.google.com by `ClovisSe...@gmail.com` on 28 Sep 2014 at 12:21

**Comment by smillst** [comment]
Closed via d34886d
